crud/common/sharding.lua

```diff
@@ -51,4 +73,94 @@ function sharding.tuple_set_and_return_bucket_id(tuple, space, specified_bucket_
     return bucket_id
 end
 
+-- Get sharding key (actually field names) for a certain space.
+-- Returns a table with field names or nil.
+function sharding.get_ddl_sharding_key(space_name, conn)
+    dev_checks('string', 'table')
+
+    if ddl_sharding_keys_cache == nil then
+        ddl_sharding_keys_cache = utils.fetch_ddl_sharding_key(conn)
+    end
+
+    return ddl_sharding_keys_cache[space_name]
+end
+
+-- Make sure fields used in sharding key are present in primary index.
+function sharding.is_sharding_key_in_primary_index(space_name, primary_index, fieldnos)
+    dev_checks('string', 'table', 'table')
+
+    if sharding_key_in_primary_index_cache[space_name] == nil then
+        sharding_key_in_primary_index_cache[space_name] = true
+        local primary_index_fieldno_map = utils.get_index_fieldno_map(primary_index)
+        for _, fieldno in ipairs(fieldnos) do
+            if primary_index_fieldno_map[fieldno] == nil then
+                sharding_key_in_primary_index_cache[space_name] = false
+                break
+            end
+        end
+    end
+
+    return sharding_key_in_primary_index_cache[space_name]
+end
+
+function sharding.schema_reload_actions(space_name, sharding_key)
+    dev_checks('string', 'table')
+
+    ddl_sharding_keys_cache[space_name] = sharding_key
+    sharding_key_in_primary_index_cache[space_name] = nil
+    sharding_key_fieldnos_cache[space_name] = nil
+    sharding_key_fieldno_map_cache[space_name] = nil
```

Comment:
I think you don't need both of them.

With idea described here (https://github.com/tarantool/crud/pull/181/files#r686230852) you will need only one auxiliary structure.

Reply:
`sharding_key_fieldno_map_cache` ha gone